packages/core/collections/tags.js

```diff
@@ -72,15 +72,19 @@ export default class Tags extends Collection {
     });
   }
 
-  async rename(tagId, newName) {
+  async rename(tagId, newName, isColorTag) {
     let tag = this.tag(tagId);
     if (!tag) {
       console.error(`No tag found. Tag id:`, tagId);
       return;
     }
+    let sanitizedName = this.sanitize(newName);
+    if (!sanitizedName)
+      throw new Error(`${isColorTag ? "Color" : "Tag"} title cannot be empty.`);
 
-    newName = this.sanitize(newName);
-    if (!newName) throw new Error("Tag title cannot be empty.");
+    if (!isColorTag) {
+      newName = sanitizedName;
+    }
```

Comment:
Just don't do any sanitization on rename. There's no need to introduce a new parameter.

Reply:
Sanitization is also required for renaming color as if we don't the user can save a color name with only spaces like "        " which will not be good.

Reply:
> which will not be good

Yeah I don't think that's an issue. If a user does that they can rename again. Renaming should allow full freedom.

Reply:
Ok